From 59b7f2816794f9537fc58f5155328349aa1d2971 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Thu, 10 Aug 2023 12:31:06 -0400
Subject: [PATCH] arch: arm64: dts: amlogic: meson gx

Le Potato: Change model name
Le Potato: Increase the max sdcard freq
Meson GX: Add power management and vrtc

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 arch/arm64/boot/dts/amlogic/meson-gx.dtsi                | 9 +++++++++
 .../boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts    | 4 ++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
index 80d86780cb6b..04fc407a0abd 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -221,6 +221,10 @@ sm: secure-monitor {
 		};
 	};
 
+	system-suspend {
+		compatible = "amlogic,meson-gx-pm";
+	};
+
 	efuse: efuse {
 		compatible = "amlogic,meson-gx-efuse", "amlogic,meson-gxbb-efuse";
 		#address-cells = <1>;
@@ -459,6 +463,11 @@ clkc_AO: clock-controller {
 				};
 			};
 
+			vrtc: rtc@a8 {
+				compatible = "amlogic,meson-vrtc";
+				reg = <0x0 0x000a8 0x0 0x4>;
+			};
+
 			cec_AO: cec@100 {
 				compatible = "amlogic,meson-gx-ao-cec";
 				reg = <0x0 0x00100 0x0 0x14>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
index 82bfabfbd39c..006b64dbcbcb 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
@@ -15,7 +15,7 @@
 / {
 	compatible = "libretech,aml-s905x-cc", "amlogic,s905x",
 		     "amlogic,meson-gxl";
-	model = "Libre Computer AML-S905X-CC";
+	model = "Libre Computer Le Potato";
 
 	aliases {
 		serial0 = &uart_AO;
@@ -308,7 +308,7 @@ &sd_emmc_b {
 
 	bus-width = <4>;
 	cap-sd-highspeed;
-	max-frequency = <50000000>;
+	max-frequency = <100000000>;
 	disable-wp;
 
 	cd-gpios = <&gpio CARD_6 GPIO_ACTIVE_LOW>;
-- 
2.39.2

From ecfe4a20b6ce46d7c376992f3a12473dd6f3b7bf Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Sun, 21 Jan 2024 21:00:59 -0500
Subject: [PATCH] arch: arm64: dts: amlogic: meson-gxl-s905x-libretech-cc: i2c_AO

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 .../boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts     | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
index 006b64dbcbcb..0bdea0b4ec07 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905x-libretech-cc.dts
@@ -20,6 +20,7 @@ / {
 	aliases {
 		serial0 = &uart_AO;
 		ethernet0 = &ethmac;
+		i2c1 = &i2c_AO;
 	};
 
 	dio2133: analog-amplifier {
@@ -342,6 +343,13 @@ &uart_AO {
 	pinctrl-names = "default";
 };
 
+&i2c_AO {
+	/* Pins 3 (SDA), 5 (SCL) */
+	status = "disabled";
+	pinctrl-0 = <&i2c_ao_pins>;
+	pinctrl-names = "default";
+};
+
 &usb {
 	status = "okay";
 	dr_mode = "host";
-- 
2.39.2

